python/ray/experimental/workflow/common.py

```diff
@@ -80,6 +83,33 @@ class WorkflowInputs:
     workflow_refs: List[WorkflowRef]
 
 
+@ray.remote
+def _hash(obj: Any) -> bytes:
+    """
+    Calculates a sha256 hash of an object.
+    """
+    # TODO (Alex): Ideally we shouldn't let ray serialize obj to begin with.
+    # (1) It would save us an unnecessary serialization/deserialization. (2)
+    # Cloudpickle doesn't isn't always stable, so for some objects (i.e.
+    # functions) this could produce inconsistent results.
+    m = hashlib.sha256()
+    # TODO (Alex): We should handle the nested ObjectRef case. This naive
+    # cloudpickle.dumps will return different a different hash when run on the
+    # recovered version of an object.
+    m.update(cloudpickle.dumps(obj))
```

Comment:
A little bit concern about memory usage. But I think we probably can investigate this later. I think in the end, we'll just get the content hash from the object store directly.